ironic/tests/unit/drivers/modules/drac/test_inspect.py

```diff
@@ -378,3 +388,176 @@ def test__get_pxe_dev_nics_with_empty_list(self, mock_get_drac_client):
+            'Location': '/redfish/v1/TaskService/Tasks/JID_704694755409'}}
+        mock_get.return_value.__dict__ = {
+            '_content': '<SystemConfiguration Model="PowerEdge R640" '
+            'ServiceTag="7T1NHK2" TimeStamp="Wed Oct 16 11:38:17 2019"> \n '
+            '<Component FQDD="NIC.Slot.1-1-1"> \n '
+            '<Attribute Name="LegacyBootProto">NONE</Attribute> \n '
+            '</Component> \n <Component FQDD="NIC.Integrated.1-1-1"> \n '
+            '<Attribute Name="LegacyBootProto">PXE</Attribute> \n '
+            '</Component> \n </SystemConfiguration> '}
+
+        with task_manager.acquire(self.context, self.node.uuid,
+                                  shared=True) as task:
+            task.driver.inspect.inspect_hardware(task)
+            pxe_dev_nics = task.driver.inspect._get_pxe_dev_nics(task)
+            self.assertEqual(expected_pxe_nic, pxe_dev_nics)
+
+    @mock.patch.object(requests, 'get', autospec=True)
+    @mock.patch.object(requests, 'post', autospec=True)
+    @mock.patch.object(redfish_utils, 'get_system', autospec=True)
+    @mock.patch.object(DracRedfishInspect, '_get_mac_address', autospec=True)
+    def test__get_pxe_dev_nics_with_BIOS_boot_mode_timeout_fail(
+            self, mock__get_mac_address, mock_get_system, mock_post, mock_get):
+        system_mock = self.init_system_mock(mock_get_system.return_value)
+        system_mock.boot.mode = 'bios'
+        mock__get_mac_address.return_value = '24:6E:96:70:49:00'
+        mock_post.return_value.__dict__ = {'status_code': 202, 'headers': {
+            'Location': '/redfish/v1/TaskService/Tasks/JID_704694755409'}}
+        mock_get.return_value.__dict__ = {}
+
+        with task_manager.acquire(self.context, self.node.uuid,
+                                  shared=True) as task:
+            self.assertRaises(exception.HardwareInspectionFailure,
+                              task.driver.inspect._get_pxe_dev_nics, task)
+
+    @mock.patch.object(redfish_utils, 'get_system', autospec=True)
+    @mock.patch.object(DracRedfishInspect, '_get_mac_address', autospec=True)
+    def test__get_pxe_dev_nics_without_boot_mode(
+            self, mock__get_mac_address, mock_get_system):
+        system_mock = self.init_system_mock(mock_get_system.return_value)
+        system_mock.boot.mode = None
+        expected_pxe_nic = []
+        mock__get_mac_address.return_value = None
+
+        with task_manager.acquire(self.context, self.node.uuid,
+                                  shared=True) as task:
+            task.driver.inspect.inspect_hardware(task)
+            pxe_dev_nics = task.driver.inspect._get_pxe_dev_nics(task)
+            self.assertEqual(expected_pxe_nic, pxe_dev_nics)
+
+    @mock.patch.object(DracRedfishInspect, '_get_pxe_dev_nics', autospec=True)
+    @mock.patch.object(redfish_utils, 'get_system', autospec=True)
+    def test_inspect_hardware_without_pxe_dev_nics(
```

Comment:
This test mocks _get_pxe_dev_nics() above, but then for the test, calls into _get_pxe_dev_nics() below, so it isn't actually testing any of the code in inspect.py.  I think you want to call task.driver.inspect.inspect_hardware() on line 517 below instead of _get_pxe_dev_nics().

Reply:
Done